### PR TITLE
feat: Implement EventDB tests, update cat-data-service endpoints

### PR DESCRIPTION
**Pull request by Mr-Leshiy**
# Description

Addded tests for the few queries of the EventDb crate, update api for the cat-data-sevice with the latest specification

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] queries::snapshot::tests::get_voter_test
- [x] queries::snapshot::tests::get_delegator_test
- [x] service::v1::registration::tests::voter_by_latest_event_test
- [x] service::v1::registration::tests::voter_by_event_test
- [x] service::v1::registration::tests::delegations_by_latest_event_test
- [x] service::v1::registration::tests::delegations_by_event_test

### CODE REVIEW

**Review by stevenj — COMMENTED** [review]
Looks good, but...

The Ideascale importer changes.  Did you make those as part of this PR, or have they been introduced when you did a sync with later ideascale importer code than is in main?

**Comment by Mr-Leshiy** [comment]
@stevenj Going to update with the latest main right now. But how Ideascale importer affect on this PR ? I have not used anything from it.

File: src/cat-data-service/src/service/v1/registration.rs [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+            "/registration/delegations/:stake_public_key",
+            get({
+                let state = state.clone();
+                move |path| async {
+                    handle_result(delegations_by_latest_event_exec(path, state).await).await
+                }
+            }),
+        )
+        .route(
+            "/registration/delegations/:stake_public_key/:event",
```
Comment (stevenj) [inline src/cat-data-service/src/service/v1/registration.rs:37]: Same comment as above.
`:event` isn't a path parameter `eid` is an optional query parameter in the format `?eid=77`.

**Review by stevenj — CHANGES_REQUESTED** [review]
Sorry discovered issue with the endpoint declarations

File: src/cat-data-service/src/service/v1/registration.rs [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+            "/registration/voter/:voting_key",
+            get({
+                let state = state.clone();
+                move |path| async {
+                    handle_result(voter_by_latest_event_exec(path, state).await).await
+                }
+            }),
+        )
+        .route(
+            "/registration/voter/:voting_key/:event",
```
[the remark this answers is not in the document]
Comment (Mr-Leshiy) [reply src/cat-data-service/src/service/v1/registration.rs:21]: Hmm, I see.
But what do you think to update API to be OpenAPI ?
I mean I have implemented it in that way, that we have actually two endpoints.
- `/registration/voter/:voting_key`, without provided event id, so using `latest` event
- `/registration/voter/:voting_key/:event`, with event id

**Review by saibatizoku — APPROVED** [review]
LGTM

**Pull request merged**